MekHQ/src/mekhq/campaign/autoResolve/AutoResolveGame.java

```diff
@@ -0,0 +1,710 @@
+            return null;
+        }
+    }
+
+    public boolean areHostile(AcsFormation formation, Player player) {
+        return player.isEnemyOf(getPlayer(formation.getOwnerId()));
+    }
+
+    // check current turn, phase, formation
+    private boolean isEligibleForAction(AcsFormation formation) {
+        return (getTurn() instanceof AcsFormationTurn)
+            && getTurn().isValidEntity(formation, this);
+    }
+
+    /**
+     * @return the first formation in the list of formations that is alive and
+     *         eligible for the current game phase.
+     */
+    public Optional<AcsFormation> getNextEligibleFormation() {
+        return getNextEligibleFormation(BTObject.NONE);
+    }
+
+    /**
+     * @return the preceding formation in the list of formations that is alive and
+     *         eligible for the current game phase.
+     */
+    public Optional<AcsFormation> getPreviousEligibleFormation() {
+        return getPreviousEligibleFormation(BTObject.NONE);
+    }
+
+    /**
+     * @return the next in the list of formations that is alive and eligible for the
+     *         current game phase,
+     *         counting from the given current formation id. If no matching
+     *         formation can be found for the given id,
+     *         returns the first eligible formation in the list of formations that
+     *         is alive and eligible.
+     */
+    public Optional<AcsFormation> getNextEligibleFormation(int currentFormationID) {
+        return getEligibleFormationImpl(currentFormationID, phase, SelectDirection.NEXT_UNIT);
+    }
+
+    /**
+     * @return the previous in the list of formations that is alive and eligible for
+     *         the current game phase,
+     *         counting from the given current formation id. If no matching
+     *         formation can be found for the given id,
+     *         returns the last eligible formation from the list of formations that
+     *         is alive and eligible.
+     */
+    public Optional<AcsFormation> getPreviousEligibleFormation(int currentFormationID) {
+        return getEligibleFormationImpl(currentFormationID, phase, SelectDirection.PREVIOUS_UNIT);
+    }
+
+    /**
+     * Based on the given search direction, returns the formation that precedes or
+     * follows the given
+     * formation ID in the list of active formations eligible for action in the
+     * given game phase.
+     *
+     * @param currentFormationID the start point of the formation search. Need not
+     *                           match an actual formation
+     * @param phase              the phase to check
+     * @param direction          the selection to seek the next or previous
+     *                           formation
+     * @return the formation that precedes or follows the given formation ID, if one
+     *         can be found
+     */
+    private Optional<AcsFormation> getEligibleFormationImpl(int currentFormationID, GamePhase phase,
```

Comment:
Seeing QL picking up a couple of 'useless params' - false positive, or something we can look at trimming?